src/backend/mpi/swdsm.cpp

```diff
@@ -1056,16 +1043,15 @@ void argo_initialize(unsigned long long size){
 	initmpi();
 	unsigned long alignment = pagesize*CACHELINE*numtasks;
 	if((size%alignment)>0){
+		size += alignment - 1;
```

Comment:
see a446f2716d43d1e65d71b3ebf06fba3d92e6d17e, this reduces the over-estimation of required memory here.